activerecord/lib/active_record/connection_adapters/abstract/transaction.rb

```diff
@@ -147,6 +151,8 @@ def initialize(connection)
       end
 
       def begin_transaction(options = {})
+        options[:run_callbacks] ||= !current_transaction.joinable?
```

Comment:
-1 to expose it. This mutates `options` argument as well.

Would consider passing a separate keyword argument instead, e.g.:
```ruby
RealTransaction.new(@connection, options, run_commit_callbacks: !current_transaction.joinable?)
```